Interop/ModelCard/src/main/java/org/tribuo/interop/modelcard/ModelCardCLI.java

```diff
@@ -0,0 +1,252 @@
+        for (int i = 0; i < considerations.size(); i++) {
+            System.out.println("\t" + i + ") "+ considerations.get(i));
+        }
+        return("Displayed all added considerations.");
+    }
+
+    @Command(
+            usage = "<String> Adds relevant factor to list of factors."
+    )
+    public String addFactor(CommandInterpreter ci, String factor) {
+        factors.add(factor);
+        return("Added factor to list of factors.");
+    }
+
+    @Command(
+            usage = "<int> Remove factor at specified index (0-indexed)."
+    )
+    public String removeFactor(CommandInterpreter ci, int index) {
+        factors.remove(index);
+        return("Removed factor at specified index.");
+    }
+
+    @Command(
+            usage = "Displays all added factors."
+    )
+    public String viewFactors(CommandInterpreter ci) {
+        for (int i = 0; i < factors.size(); i++) {
+            System.out.println("\t" + i + ") "+ factors.get(i));
+        }
+        return("Displayed all added factors.");
+    }
+
+    @Command(
+            usage = "<String> Adds resource to list of resources."
+    )
+    public String addResource(CommandInterpreter ci, String resource) {
+        resources.add(resource);
+        return("Added resource to list of resources.");
+    }
+
+    @Command(
+            usage = "<int> Remove resource at specified index (0-indexed)."
+    )
+    public String removeResource(CommandInterpreter ci, int index) {
+        resources.remove(index);
+        return("Removed resource at specified index.");
+    }
+
+    @Command(
+            usage = "Displays all added resources."
+    )
+    public String viewResources(CommandInterpreter ci) {
+        for (int i = 0; i < resources.size(); i++) {
+            System.out.println("\t" + i + ") "+ resources.get(i));
+        }
+        return("Displayed all added resources.");
+    }
+
+    @Command(
+            usage = "<String> Records primary contact in case of questions or comments."
+    )
+    public String primaryContact(CommandInterpreter ci, String contact) {
+        builder.primaryContact(contact);
+        return("Recorded primary contact as " + contact + ".");
+    }
+
+    @Command(
+            usage = "<String> Records model's citation."
+    )
+    public String modelCitation(CommandInterpreter ci, String citation) {
+        builder.modelCitation(citation);
+        return("Recorded model citation as " + citation + ".");
+    }
+
+    @Command(
+            usage = "<String> Records model's license."
+    )
+    public String modelLicense(CommandInterpreter ci, String license) {
+        builder.modelLicense(license);
+        return("Recorded model license as " + license + ".");
+    }
+
+    @Command(
+            usage = "<filename> Saves UsageDetails to an existing ModelCard file and close CLI."
+    )
+    public String saveUsageDetails(CommandInterpreter ci, File destinationFile) throws IOException {
```

Comment:
Most of the other CLIs we have let you load in different files (e.g. models, datasets) to explore them without restarting the JVM to get a fresh shell. It might be nicer to have this CLI take a filename as an argument for a pre-existing model card, have it load that in on startup, and then you can display it in the CLI while you write the usage details. This method could then save out to a different filename and clear out the usage builder (by making  a fresh one or adding a clear method), clear all the lists, and then the user can load in another model card to write the next set of usage details.

Reply:
Done. The user now has the command to start a newUsageDetails, which will effectively allow them to write the UsageDetails for another model card.